pkg/kubectl/cmd/set/set_port.go

```diff
@@ -0,0 +1,281 @@
+	cmd.Flags().Int("port", 80, i18n.T("Name or number for the port on the container that the service should direct traffic to. Optional."))
+	cmd.Flags().String("protocol", "tcp", i18n.T("The network protocol for the service to be created. Default is 'TCP'."))
+	cmd.Flags().BoolVar(&options.overwrite, "overwrite", true, "If true, allow environment to be overwritten, otherwise reject updates that overwrite existing environment.")
+	cmdutil.AddRecordFlag(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+	return cmd
+}
+
+func (o *PortOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.Mapper, o.Typer = f.Object()
+	o.UpdatePodSpecForObject = f.UpdatePodSpecForObject
+	o.portsForObject = f.PortsForObject
+	o.Encoder = f.JSONEncoder()
+	o.ShortOutput = cmdutil.GetFlagString(cmd, "output") == "name"
+	o.Record = cmdutil.GetRecordFlag(cmd)
+	o.ChangeCause = f.Command(cmd, false)
+	o.PrintObject = f.PrintObject
+	o.Local = cmdutil.GetFlagBool(cmd, "local")
+	o.DryRun = cmdutil.GetDryRunFlag(cmd)
+	o.Output = cmdutil.GetFlagString(cmd, "output")
+	o.port = cmdutil.GetFlagInt(cmd, "port")
+	o.protocol = strings.ToUpper(cmdutil.GetFlagString(cmd, "protocol"))
+	o.containerSelector = cmdutil.GetFlagString(cmd, "containers")
+	o.overwrite = cmdutil.GetFlagBool(cmd, "overwrite")
+	o.Cmd = cmd
+
+	cmdNamespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	builder := f.NewBuilder(!o.Local).
+		ContinueOnError().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		Flatten()
+	if !o.Local {
+		builder = builder.
+			SelectorParam(o.Selector).
+			ResourceTypeOrNameArgs(o.All, args...).
+			Latest()
+	}
+	o.Infos, err = builder.Do().Infos()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *PortOptions) Validate() error {
+	errors := []error{}
+	if o.protocol != "TCP" && o.protocol != "UDP" {
+		errors = append(errors, fmt.Errorf("protocol must tcp or udp"))
+	}
+	if o.port < 0 {
+		errors = append(errors, fmt.Errorf("port must greater than 0"))
+	}
+	return utilerrors.NewAggregate(errors)
+}
+
+func (o *PortOptions) Run() error {
+	allErrs := []error{}
+
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) ([]byte, error) {
+		transformed := false
+		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *api.PodSpec) error {
+			containers, _ := selectContainers(spec.Containers, o.containerSelector)
+			if len(containers) == 0 {
+				fmt.Println(info.Mapping.Resource, info.Name, o.containerSelector)
+				_, err := fmt.Fprintf(o.Err, "warning: %s/%s does not have any containers matching %q\n", info.Mapping.Resource, info.Name, o.containerSelector)
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+
+			ports, err := o.portsForObject(info.Object)
+			if err != nil {
+				return err
+			}
+			for _, c := range containers {
+				if !o.overwrite {
```

Comment:
L199-L205 seems have nothing to do with `c`. Can we move it to outer?

Reply:
yes, thank you point it